swift/test/Ice/ami/AllTests.swift

```diff
@@ -92,122 +89,171 @@ func allTests(_ helper: TestHelper, collocated: Bool = false) throws {
         let i = p.ice_adapterId("dummy")
 
         do {
-            _ = try i.ice_isAAsync(id: "::Test::TestIntf").wait()
+            _ = try await i.ice_isAAsync(id: "::Test::TestIntf")
             try test(false)
         } catch is Ice.NoEndpointException {}
 
         do {
-            try i.opAsync().wait()
+            try await i.opAsync()
             try test(false)
         } catch is Ice.NoEndpointException {}
 
         do {
-            _ = try i.opWithResultAsync().wait()
+            _ = try await i.opWithResultAsync()
             try test(false)
         } catch is Ice.NoEndpointException {}
 
         do {
-            try i.opWithUEAsync().wait()
+            try await i.opWithUEAsync()
             try test(false)
         } catch is Ice.NoEndpointException {}
 
         // Ensures no exception is called when response is received
-        _ = try p.ice_isAAsync(id: "::Test::TestIntf").wait()
-        try p.opAsync().wait()
-        _ = try p.opWithResultAsync().wait()
+        _ = try await p.ice_isAAsync(id: "::Test::TestIntf")
+        try await p.opAsync()
+        _ = try await p.opWithResultAsync()
 
         do {
             // If response is a user exception, it should be received.
-            try p.opWithUEAsync().wait()
+            try await p.opWithUEAsync()
             try test(false)
         } catch is TestIntfException {}
     }
     output.writeLine("ok")
 
     output.write("testing sent callback... ")
-    _ = try Promise<Bool> { seal in
-        _ = p.ice_isAAsync(id: "") { sentSynchronously in
-            seal.fulfill(sentSynchronously)
-        }
-    }.wait()
-
-    _ = try Promise<Bool> { seal in
-        _ = p.ice_pingAsync { sentSynchronously in
-            seal.fulfill(sentSynchronously)
-        }
-    }.wait()
 
-    _ = try Promise<Bool> { seal in
-        _ = p.ice_idAsync { sentSynchronously in
-            seal.fulfill(sentSynchronously)
+    await withCheckedContinuation { continuation in
+        Task {
+            do {
+                _ = try await p.ice_isAAsync(id: "") { sentSynchronously in
```

Comment:
It's not clear what you're testing here. That the continuation is called?

Reply:
Yes.